cli/pkg/kctrl/cmd/package/builder/package-build.go

```diff
@@ -0,0 +1,38 @@
+package builder
+
+import (
+	"github.com/vmware-tanzu/carvel-kapp-controller/pkg/apiserver/apis/datapackaging/v1alpha1"
+	vendirconf "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type PackageBuild struct {
+	metav1.TypeMeta `json:",inline"`
+	Spec            Spec `json:"spec, omitempty"`
+}
+
+type Imgpkg struct {
+	metav1.TypeMeta  `json:",inline"`
+	RegistryURL      string `json:"registryUrl"`
+	RegistryUserName string `json:"registryUserName"`
+	RegistryPassword string `json:"registryPassword"`
+	RepoName         string `json:"repoName"`
+	Tag              string `json:"tag"`
+}
+
+type Spec struct {
+	Pkg         v1alpha1.Package         `json:"package"`
+	PkgMetadata v1alpha1.PackageMetadata `json:"packageMetadata"`
+	Vendir      vendirconf.Config        `json:"vendir"`
+	Imgpkg      Imgpkg                   `json:"imgpkg"`
```

Comment:
all of these fields should be pointers to indicate that they are optional. also marked as omitempty.

Reply:
Adopted